src/main.py

```diff
@@ -30,14 +31,66 @@ def handle_invalid_usage(error):
 def sitemap():
     return generate_sitemap(app)
 
-@app.route('/user', methods=['GET'])
-def handle_hello():
+@app.route('/users/<int:id>', methods=['GET'])
+def get_user(id):
+    if User.get_user(id) is None:
+        raise APIException('User not found', status_code=404)
+    else:
+        return User.get_user(id)
+
+@app.route('/login', methods=['GET'])
+def get_user_by_email():
+    body=request.get_json()
+
+    if User.get_user_by_email(body["email"],body["password"]) is "email":
+        raise APIException("email doesn't exist", status_code=404)
+    elif User.get_user_by_email(body["email"],body["password"]) is "password":
+        raise APIException("password for that email doesn't exist", status_code=404)
+    else:
+        return User.get_user_by_email(body["email"],body["password"])
```

Comment:
```suggestion
    user = User.get_user_by_email(body["email"],body["password"])
    if not user:
        raise APIException("Login failed", status_code=401)
   return user
```
Varios puntos aquí:

1. El primero es que se repiten peticiones que se pueden guardar en una variable y luego reutilizar el resultado, con la sugerencia que he añadido en un login válido se hacen 2 hits menos a la base de datos.
2. Este cambio está relacionado con el que comenté en la reunión del viernes hablando de que si no encuentras un usuario o el usuario no tiene un login válido lo normal es devolver un valor vacío en vez de un string. Para aceptar este cambio es necesario aceptar también el relacionado en el modelo de `User`.
3. Por motivos de privacidad no se debe indicar si un usuario tiene o no cuenta a alguien que no conozca la contraseña de login, por eso he cambiado el error